File: RayTracer/Transform/Scaling.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace RayTracer
{
        [Serializable]
    public class Scaling : Transform
    {

        public Scaling()
            : this(new Point3(1,1,1))
        { }

        public Scaling(Double[] values)
            : this(new Point3(values))
        { }

        public Scaling(Point3 values)
        {
            Matrix = new MyMatrix(4, 4);

            Matrix.SetValue(0, 0, values.X);
            Matrix.SetValue(1, 1, values.Y);
            Matrix.SetValue(2, 2, values.Z);

            Matrix.SetValue(3, 3, 1f);
           // ShowInformation();
        }

       
    }
}

File: RayTracer - BVH - CUDA/RayTracerCUDA/Scaling.cpp
#include "Scaling.h"



Scaling::Scaling(float x, float y, float z)
{
	matrix = Matrix(4, 4);
	matrix(0, 0) = x;
	matrix(1, 1) = y;
	matrix(2, 2) = z;
	matrix(3, 3) = 1.f;

}

Scaling::~Scaling()
{
}
